### PR TITLE
New programme and cohort content entities, with config

### PR DESCRIPTION
**Pull request by joesb**
[ch474]
[ch476]
[ch481]
[ch481]
[ch251]
[ch253]

This adds the programme year and cohort entity types to the system. Cohorts have an optional node entity relationship to programme years. Cohort name fields are generated automatically from the cohort number

### CODE REVIEW

File: config/sync/block.block.help.yml [hunk 1]
```diff
@@ -0,0 +1,20 @@
+uuid: 647a802d-5a41-48ce-a724-c849c1b7466e
```
Comment (DarkSnow) [inline config/sync/block.block.help.yml]: Is this block relevant to the PR? Not a problem if it isn't, especially given that it's for the admin theme.

Comment (joesb) [reply config/sync/block.block.help.yml]: Right, yes. I added this block because I added some help to the Cohort node (cf. `config/sync/node.type.cohort.yml`) that wasn't being displayed on node create/edit forms. Hence switching on Help module and adding this block. 

Originally I used [serial](https://www.drupal.org/project/serial) contrib module so the cohort number was being incremented automatically by the system without need for human intervention. I then removed it because, of course, the 'cohort number' field will be synchronised with Salesforce in due course — Salesforce will define the number assigned to the cohort (presumably from a serial/auto-incrementing field there) so there's not the same need for this help to display. I can remove it, and switch off Help module.

Comment (joesb) [reply config/sync/block.block.help.yml]: Done now, in be55f9a
